crates/runtime/src/dataconnector/odbc.rs

```diff
@@ -176,12 +176,54 @@ fn driver_is_file(driver: &str) -> bool {
         .is_some()
 }
 
+macro_rules! parameter_is_integer {
+    ($param:expr) => {
+        let value = $param
+            .parse::<usize>()
+            .map_err(|_| Error::InvalidParameter {
+                param: stringify!($param).to_string(),
+                msg: "The parameter must be a positive integer.".to_string(),
+            })?;
+
+        if value == 0 {
+            return Err(Error::InvalidParameter {
+                param: stringify!($param).to_string(),
+                msg: "The parameter must be a positive integer.".to_string(),
+            }
+            .into());
+        }
+    };
+}
+
 impl DataConnectorFactory for ODBCFactory {
     fn create(
         &self,
         params: DataConnectorParams,
     ) -> Pin<Box<dyn Future<Output = super::NewDataConnectorResult> + Send>> {
         Box::pin(async move {
+            if let Some(binary_size) = params.parameters.get("max_binary_size").expose().ok() {
+                parameter_is_integer!(binary_size);
```

Comment:
I'm not a big fan of macros that obscure the control flow - I would prefer the macro just return a Result that can be returned directly here.